<a href="https://colab.research.google.com/github/carlosvasquez3/M.Sc.-in-Data-Science/blob/main/Proyecto_Unidad2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#TITANIC


El dataset contiene información de pasajeros del Titanic, incluyendo características demográficas, socioeconómicas y de viaje, junto con una variable objetivo que indica si sobrevivieron o no al naufragio.

hola

#Proyecto

##Librerías


In [ ]:
# Importe de librerías
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Carga de datos

In [ ]:
# Se crea el dataframe
df = pd.read_csv('https://raw.githubusercontent.com/carlosvasquez3/M.Sc.-in-Data-Science/refs/heads/main/Programaci%C3%B3n%20b%C3%A1sica/Data/train.csv') #Se utiliza github para almacenar la data
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

Se cargó el dataset desde GitHub usando su enlace raw y se verificó la correcta importación mostrando las primeras filas del DataFrame. Se procede a realizar el perfilado de datos.

## Perfilado de datos

Significado de las variables de la data.

| Variable   | Definición                                                | Clave |
|------------|-----------------------------------------------------------|-------|
| survival   | Supervivencia                                             | 0 = No, 1 = Sí |
| pclass     | Clase del boleto                                          | 1 = Primera, 2 = Segunda, 3 = Tercera |
| sex        | Sexo del pasajero                                         | — |
| age        | Edad en años                                              | — |
| sibsp      | Número de hermanos / cónyuges a bordo                    | — |
| parch      | Número de padres / hijos a bordo                          | — |
| ticket     | Número de boleto                                          | — |
| fare       | Tarifa pagada por el pasajero                             | — |
| cabin      | Número de cabina                                          | — |
| embarked   | Puerto de embarque                                        | C = Cherbourg, Q = Queenstown, S = Southampton |


In [ ]:
# Información de las variables
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


El dataset contiene 891 registros y 12 variables. Se identifican variables numéricas (int64 y float64) y variables categóricas. Se observa que la columna Age presenta valores nulos, al igual que Cabin y Embarked.

In [ ]:
# Resumen estadistico
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

El resumen estadístico muestra que la mayoría de los pasajeros viajaban en tercera clase y que aproximadamente el 38% sobrevivió. La edad promedio es cercana a los 30 años, aunque presenta valores faltantes. La mayoría viajaba sin familiares a bordo y la tarifa pagada presenta una alta variabilidad, con valores máximos considerablemente elevados.

In [ ]:
# Nulos
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

La variable Edad presenta 177 valores faltantes, mientras que las variables Género (Sex) y Sobreviviente (Survived) no presentan valores nulos.

## Preprocesamiento de datos

Se eliminó la variable Cabin debido a la gran cantidad de valores nulos que contiene y a que no es necesaria para el análisis. Esto permite conservar más registros en el proceso de limpieza.

In [ ]:
# Se elimina la variable Cabin del modelo
df = df.drop(columns=['Cabin'])

In [ ]:
# Leer el nuevo dataframe
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Embarked  
0      0         A/5 21171   7.2500        S  
1      0          PC 17599  71.2833        C  
2      0  STON/O2. 3101282   7.9250        S  
3      0            113803  53.1000        S  
4      0            373450   8.0500        S

Variable eliminada con exito

In [ ]:
# Se veriifica nuevamente los nulos
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Embarked         2
dtype: int64

In [ ]:
# Se eliminan los registros con valores nulos
df = df.dropna()
df.shape

(712, 11)

In [ ]:
# Se veriifica nuevamente los nulos
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

Después de eliminar la variable Cabin, se verificaron nuevamente los valores nulos y se procedió a eliminar los registros restantes con datos faltantes, tal como lo indica el proyecto. Es importante resaltar que los datos son un recurso valioso, por lo que la eliminación de registros debe realizarse con precaución, ya que puede reducir la muestra y afectar los resultados del análisis. En este caso, se tomó la decisión para garantizar un conjunto de datos limpio y consistente. **Por objeto de estudio**

In [ ]:
df['Familia'] = df['SibSp'] + df['Parch'] + 1        #El tamaño de la familia del pasajero.


In [ ]:
df.loc[:,'Sex'] = df['Sex'].str.upper()      #Estandarización para la columna Sex
df.loc[:, 'Sex'] = df['Sex'].replace({       #Conversión a valores reales
    'FEMALE': 1,
    'MALE': 2,                               #Ahora la variable de sexo femenino equivale a 1 y masculino equivale a 2
    'F': 1,
    'M': 2
})
df

In [ ]:
df.loc[:,'Embarked'] = df['Embarked'].str.upper()      #Estandarización para la columna Embarked
df.loc[:, 'Embarked'] = df['Embarked'].replace({       #Conversión a valores reales
    'C': 1,
    'Q': 2,                                            # Ahora la variable Cherbourg equivale a 1, Queenstown equivale a 2 y Southampton equivale a 3
    'S': 3,
})
df